doc/async_worker.md

```diff
@@ -1,5 +1,192 @@
-# Async worker
+# AsyncWorker
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+`AsyncWorker` is an abstract class that you can subclass to remove much of the
+tedious tasks on moving data between the event loop and worker threads. This
+class internally handles all the details of creating and executing asynchronous
+operation.
+
+## Methods
+
+### Env
+
+Requests the environment in which the async worker has been initially created.
+
+```cpp
+Env Env() const;
+```
+
+Returns the environment in which the async worker has been created.
+
+### Queue
+
+Requests that the created work or task will be placed on the queue of the execution.
```

Comment:
Perhaps instead of "placed on the queue of the execution" we could say "queued for execution".